_lessons/python-scatter/06-profiling_with_MAP.md

```diff
@@ -1,77 +1,79 @@
+* ensure the test case is representative of a production run
 
-**Note:** keep in mind, that with shortened computation time, the initialisation and finalisation becomes more dominant.
+**Note:** keep in mind, that with shortened computation time, the initialisation and finalisation steps may become dominant.
 
 
-## ARM MAP profiler
+## MAP profiler
 
-On NeSI systems the [ARM MAP](https://www.arm.com/products/development-tools/server-and-hpc/forge/map) profiler is provided as part of the `forge` module (along with the parallel debugger, DDT).
+On NeSI systems the [ARM MAP](https://www.arm.com/products/development-tools/server-and-hpc/forge/map) profiler is provided as part of the *forge* module (along with the parallel debugger DDT).
 
-MAP is a commercial product, which can profile parallel, multi-threaded and single-threaded C/C++, Fortran and F90, as well as Python codes. It can be used without code modification.
-MAP can be launched with a GUI and without. The GUI allows the user to navigate through the code and enables them to focus on specific source lines. The "Express Launch", without GUI enables easy usage of existing queue scripts and workflows.
+MAP is a commercial product, which can profile parallel, multi-threaded and single-threaded C/C++, Fortran, as well as Python code. It can be used without code modification.
+MAP can be launched with a graphical user interface and without. The graphical user interface allows the user to navigate through the code and focus on specific source lines. The "Express Launch", without graphical user interface, makes it ieasy to submit job scripts and workflows.
 
 For more details see the [ARM MAP documentation](https://developer.arm.com/docs/101136/latest/map).
 
-In the following, both GUI and express launch versions are used with the scatter example.
+In the following, both the graphical user interface and express launch versions are used with the scatter example.
 
-## Scatter MPI test case
+## Code example
 
-We'll use the code in directory `mpi`. Start by
+We'll use the *scatter.py* code in directory `mpi` of the *solutions* branch. Start by
 
 ```
+git --fetch all
+git checkout solutions
 cd mpi
 ```
 
-The MPI version of scatter performs the default test case within a short time (less than 20 seconds). We could argue that this is too short. Thus, we increase the problem size:
-```
-python scatter.py -c -nx 256 -ny 256 -nc 256
-```
 ## Using the "Express Launch"
 
-To use ARM MAP we need to load `module load forge` in our batch script and add `map --profile` in front of the parallel run statements. Thus, `srun` and its options, as well as our executable and its arguments are passed to MAP.
-Thus in the script we will have something like:
-
+To use MAP we need to load the *forge* module in our batch script and add `map --profile` in front of the parallel run statements. For example:
```

Comment:
To be consistent I would also here write "To use MAP we need to load `module load forge` in our batch script ..."